### PR TITLE
Added more exceptions to SqlRetry

### PR DESCRIPTION
**Pull request by shuhaowu**
Also `retriable` is correctly passed to SqlRetry instances.

### CODE REVIEW

**Comment by shuhaowu** [comment]
Test passes locally. No additional tests are added because it's not obvious how to test this. 

Do I need to bump the version in this PR as well? I'm also no famliar with how to deploy an updated gem.

```
............
Finished in 0.237227s, 429.9679 runs/s, 834.6436 assertions/s.
102 runs, 198 assertions, 0 failures, 0 errors, 0 skips
[cut]
....
Finished in 56.590976s, 1.2723 runs/s, 2.9863 assertions/s.
72 runs, 169 assertions, 0 failures, 0 errors, 0 skips

```

File: lib/lhm/sql_retry.rb [hunk 1]
```diff
@@ -39,6 +39,8 @@ def default_retry_config
             /Lock wait timeout exceeded/,
             /Timeout waiting for a response from the last query/,
             /Deadlock found when trying to get lock/,
+            /Query execution was interrupted/,
+            /Lost connection to MySQL server during query/,
           ]
```
Comment (jeremycole) [inline lib/lhm/sql_retry.rb]: Maybe should also handle this one that we see recently, too?

```suggestion
            /Max connect timeout reached/,
          ]
```

Comment (shuhaowu) [reply lib/lhm/sql_retry.rb]: Oh I must have missed it in those screenshots. Updated.

**Comment by shuhaowu** [comment]
So the errors we were seeing are coming from methods like `upper_id` and what not. I can hack around the problem for now but to fix it comprehensively requires monkey patching `connection` (probably not a good idea), or refactoring all database interaction logic by wrapping `connection` with SqlRetry...

**Comment by shuhaowu** [comment]
I investigated a bit more, it turns out the new verifier.call within the Chunker could also cause LHM failures. I've worked around this as well in this PR and thus am adding more reviewers as it is a more substantial change.

Long term solution still requires some refactoring, but this should work for now.

File: lib/lhm/chunker.rb [hunk 2]
```diff
@@ -54,8 +60,19 @@ def bottom
       @next_to_insert
     end
 
+    def verify_can_run
+      return unless @verifier
+      @retry_helper.with_retries do |retriable_connection|
```
Comment (insom) [inline lib/lhm/chunker.rb]: 👏

**Comment by shuhaowu** [comment]
One further side effect is that the retries will make the chunker loop stall even more.

**Pull request merged**